### PR TITLE
UCT/BASE: Use uct_md_query_v2 in uct_mem_alloc and check EXPORT

### PR DESCRIPTION
**Pull request by dmitrygx**
## What

Use `uct_md_query_v2` in `uct_mem_alloc` and check `UCT_MD_FLAG_EXPORTED_MKEY` capability if `UCT_MD_MEM_FLAG_EXPORT` was asked along with memory allocation.

## Why ?

To theoretically support MDs with memory allocation and exporting memory.

## How ?

1. Use `uct_md_query_v2` instead of `uct_md_query` in `uct_mem_alloc`.
2. Check `UCT_MD_FLAG_EXPORTED_MKEY` MD attribute if `UCT_MD_MEM_FLAG_EXPORT` flag has been specified.

### CODE REVIEW

**Comment by dmitrygx** [comment]
@brminich @Artemy-Mellanox could you review pls?

File: src/uct/base/uct_mem.c [hunk 1]
```diff
@@ -120,26 +120,33 @@ ucs_status_t uct_mem_alloc(size_t length, const uct_alloc_method_t *methods,
...
                 /* Check if MD supports allocation with fixed address
                  * if it's requested */
                 if ((flags & UCT_MD_MEM_FLAG_FIXED) &&
-                    !(md_attr.cap.flags & UCT_MD_FLAG_FIXED)) {
+                    !(md_attr.flags & UCT_MD_FLAG_FIXED)) {
+                    continue;
+                }
+
+                /* Check if MD supports allocation of memory is exported if
+                 * it's requested */
```
Comment (brminich) [inline src/uct/base/uct_mem.c:142]: imo, this comment is less clear than the code below.  I'd remove it or did you mean
`Check if MD supports allocation of memory as exported if it's requested */`

File: src/uct/base/uct_mem.c [hunk 2]
```diff
@@ -120,26 +120,33 @@ ucs_status_t uct_mem_alloc(size_t length, const uct_alloc_method_t *methods,
...
+                if ((flags & UCT_MD_MEM_FLAG_EXPORT) &&
+                    !(md_attr.flags & UCT_MD_FLAG_EXPORTED_MKEY)) {
```
Comment (brminich) [inline src/uct/base/uct_mem.c:144]: do we really need to support this combination? What is the use-case?

File: src/uct/base/uct_mem.c (hunk 1 above)
Comment (dmitrygx) [reply src/uct/base/uct_mem.c]: yes, I meant using `as` instead of `is`
fixed

File: src/uct/base/uct_mem.c (hunk 2 above)
Comment (dmitrygx) [reply src/uct/base/uct_mem.c:144]: yes, we should support it for theoretical transports which could support ALLOCATE and EXPORT simultaneously.
now we don't have such transports, but users could pass `EXPORT` flag anyway and we should handle it, i.e. not allocate memory using MD and move to another method of memory allocation.

**Comment by dmitrygx** [comment]
@brminich is it ok?

**Review by brminich — APPROVED** [review]

**Comment by dmitrygx** [comment]
@yosefe could you review pls?

**Pull request closed without merging**